services/app-web/src/components/HealthPlanPackageTable/HealthPlanPackageTable.test.tsx

```diff
@@ -0,0 +1,647 @@
+            name: 'MCR-OH-0069-PMAP',
+            programs: [
+                {
+                    __typename: 'Program',
+                    id: 'abbdf9b0-c49e-4c4c-bb6f-040cb7b51cce',
+                    fullName: 'Special Needs Basic Care',
+                    name: 'SNBC',
+                },
+                {
+                    __typename: 'Program',
+                    id: 'd95394e5-44d1-45df-8151-1cc1ee66f100',
+                    name: 'PMAP',
+                    fullName: 'Prepaid Medical Assistance Program',
+                },
+                {
+                    __typename: 'Program',
+                    id: 'ea16a6c0-5fc6-4df8-adac-c627e76660ab',
+                    fullName: 'Minnesota Senior Care Plus ',
+                    name: 'MSC+',
+                },
+            ],
+            submittedAt: '2022-11-05',
+            status: 'SUBMITTED',
+            updatedAt: new Date('2022-11-05T17:47:09.745Z'),
+            submissionType: 'Contract action and rate certification',
+            stateName: 'Ohio',
+        },
+        {
+            id: '74c3c976-45d8-49fe-ac76-6ae3147acd12',
+            name: 'MCR-PR-0065-PMAP',
+            programs: [
+                {
+                    __typename: 'Program',
+                    id: 'd95394e5-44d1-45df-8151-1cc1ee66f100',
+                    name: 'PMAP',
+                    fullName: 'Prepaid Medical Assistance Program',
+                },
+            ],
+            submittedAt: '2022-10-05',
+            status: 'SUBMITTED',
+            updatedAt: new Date('2022-10-05T17:45:05.562Z'),
+            submissionType: 'Contract action and rate certification',
+            stateName: 'Puerto Rico',
+        },
+    ]
+
+    it('renders table with expected number of submissions', async () => {
+        renderWithProviders(
+            <HealthPlanPackageTable
+                tableData={submissions}
+                userType="CMSUser"
+            />
+        )
+        const rows = await screen.findAllByRole('row')
+        expect(screen.getByRole('table')).toBeInTheDocument()
+        //Expect 5 rows. 4 data rows and 1 header row
+        expect(rows).toHaveLength(5)
+    })
+
+    it('displays no submission text when no submitted packages exist', async () => {
+        renderWithProviders(
+            <HealthPlanPackageTable tableData={[]} userType="CMSUser" />,
+            {
+                apolloProvider: {
+                    mocks: [
+                        fetchCurrentUserMock({
+                            statusCode: 200,
+                        }),
+                    ],
+                },
+            }
+        )
+
+        await waitFor(() => {
+            expect(
+                screen.getByText(/You have no submissions yet/)
+            ).toBeInTheDocument()
+            expect(screen.queryByRole('table')).toBeNull()
+        })
+    })
+
+    it('displays submissions table with expected headers for cms users', async () => {
```

Comment:
suggestion: Still separate out the cms and state user specific behavior is `describe` blocks. Right now things are intermingled its a bit hard to code and review as well as read/review tests and get a mental model of the expected behavior.